main/src/library/Fixpoint/Compiler.flix

```diff
@@ -37,7 +37,7 @@ namespace Fixpoint {
         case Datalog(facts, rules) =>
             let stmts = MutList.new();
             Array.foreach(compileRule(stmts), facts);
-            Array.foldRight(rule -> match rule {
+            Array.foldRightOld(rule -> match rule {
```

Comment:
I can't figure out how to make this work with the new `Array.foldRight`.

Reply:
I think it just becomes: 

```
Array.foldRightOld(k-> match k() {
```

Does that help?

Reply:
Hmm, it does not seem to work. This still gives me an error.
```
Array.foldRight(rule -> match rule() {
    case Constraint(HeadAtom(p, _, _), _) =>
        let ruleStratum = Map.getWithDefault(p, 0, strat);
        Map.insertWith(List.append, ruleStratum, rule() :: Nil)
}, Map#{}, rules)
```
with error: `>> Unable to unify the types: '? ->{Impure} ?' and 'Map'. `

One of the function I made work e.g. `def compileRule` had a `k`
```
Array.foldRight(match (atom, rowVar) -> k -> match atom { ... })
```
I do not know, if that is the problem I'm experiencing here. That I need some `k` in the middle of `rule -> k -> match rule`

Reply:
That worked :-)